### DISCUSSION TITLE
Harvest ideas from other overhead run-and-gun games

### DISCUSSION THREAD

**Discussion by cxong** [cxong/cdogs-sdl#846] [General]
C-Dogs is an overhead run-and-gun game. It's worth looking at other games in the same genre, what they did well, what they didn't do well, what parts are fun, what parts didn't work.

**Comment by cxong** [comment]
![](http://www.hardcoregaming101.net/jackal/jackal-arcade3.png)

[Jackal](http://www.hardcoregaming101.net/jackal/jackal.htm)

Pros:
- Weapon upgrades, hostages encourage perfect play (lose them when you die)
- Weapon upgrades "expand" its capabilities, rather than change them or are complete upgrades
- Konami music
- Homing missiles
- Many enemy types (infantry, vehicles, ships, aircraft, turrets)

Cons:
- Have to wait for the jeep to turn completely before shooting in the new direction
- When scrolling ahead, very little is shown which makes it easier to run into enemies (probably intended for a quarter-muncher)

**Comment by cxong** [comment]
![](http://contra.kontek.net/games/supercnes/superc3.gif)

[Super C](http://contra.kontek.net/games/supercnes/supercnes.htm) (the overhead levels)

Pros:
- Powerups
- Asymmetrical enemy weapons (much weaker, but can fire off-axis; solves the C-Dogs same-weapon difficulty problem)
- Hit sounds (#41)
- Bosses

Cons:
- Only walking discourages fast movement (with side-scrolling you could jump/crouch to rapidly evade)
- Relatively small field of action

**Comment by cxong** [comment]
![](http://www.10tons.com/Screenshots/image/games/crimsonland/screenshots/windows/crimsonland_windows_ss_2.jpg)

[Crimsonland](http://www.crimsonland.com/)

Pros:
- Powerups, weapon pickups
- Huge range of weapons
- Hordes of varied enemies (some split into smaller enemies when killed, some spawn other enemies)

Cons:
- Repetitive - enemies just walk towards player, later levels comes down to hoping to get good powerups before dying
- Messy action
- Hard to balance the huge range of weapons; some weapons are great earlier on but useless later, and vice versa

**Comment by cxong** [comment]
![](http://videogamecritic.com/images/nes/heavy_barrel.gif)

[Heavy barrel](http://videogamecritic.com/nesfh.htm)

Two notes, not so much pros/cons as characteristics:
- Weapon powerups, give you a certain amount of ammo (think metal slug). This is interesting as it punishes bad play (lose them when you die), but usually enough ammo is given so that if you don't die you spend most of the game with powerful weapons, plus you are forced to use different ones as the ammo runs out, which adds variety. This design definitely has downsides, however.
- Grenades - interesting in theory, but in this game they weren't compelling enough

**Comment by cxong** [comment]
![](http://www.mobygames.com/images/shots/l/312327-front-line-nes-screenshot-an-enemy-gets-killed-s.gif)

[Front Line](http://www.mobygames.com/game/front-line)

Pros:
- Wacky graphics/animation
- Different weapons (grenade goes over walls)
- Tanks

Cons:
- Slow pace, hard to aim
- Unpredictable enemy attacks

**Comment by cxong** [comment]
![](http://jwtiedostot.pp.fi/pelit/kuvat__/tapan_kaikki_4_bloodshed.jpg)

[Tapan Kaikki](http://everything2.com/title/Tapan+Kaikki)

Pros:
- Effects (FOV, particles, dynamic lighting, blood, gibs, footprints)
- Fast action (bullets travel very fast)
- Buying weapons and items

Cons:
- Control scheme awkward (keyboard turn and strafe)
- Hard to aim; laser sight is an item upgrade
- Harder enemies are frustrating - need to aim and kill them in short amount of time because it's impossible to evade

**Comment by cxong** [comment]
![](http://www.abandonia.com/files/games/486/Assault%20Trooper_5.png)

[Assault trooper](http://www.abandonia.com/games/486)

Pros:
- Weapon system based on weight (combine unbalanced weapons using weight quota)
- Health pickups
- Grenade key (#33)
- Physics effects (items move when shot, can destroy fences/walls)
- Radar (#27)
- Different enemies (dogs, vehicles)

Cons:
- Movement awkward (similar to Tapan Kaikki)
- Stealth not implemented well (can hide in bushes)
- Crouching not useful enough

**Comment by cxong** [comment]
![](http://www.old-games.com/screenshot/2146-5-fire-fight.jpg)

[Fire fight](http://www.old-games.com/download/2146/fire-fight)

Pros:
- Radar (#27)
- Mission chatter
- [Parallax scrolling](http://en.wikipedia.org/wiki/Parallax_scrolling)
- Off-centre camera, shows more in front
- Interesting weapons (swarmer: homing missiles)

Cons:
- Turning is slow, worsened by gameplay/level design (often need to turn large angles)
- Combat is often up-close exchanges of fire

**Comment by cxong** [comment]
![](http://www.mobygames.com/images/shots/l/31121-urban-strike-genesis-screenshot-chasing-down-truckss.gif)

[Strike series](http://www.mobygames.com/game-group/electronic-arts-strike-series)

Most characteristics shared with Fire Fight.
- Auto-aim (helps with awkward control scheme, also included as game mechanic with better gunners)
- Fuel keeps tension up

**Comment by cxong** [comment]
![](http://www.mobygames.com/images/shots/l/555712-battle-city-nes-screenshot-enemy-destroyed-s.png)

[Battle city](http://www.mobygames.com/game/battle-city)

Pros:
- Powerups, upgrades
- Gameplay focused on destructible walls
- Ground effects (#35)

Cons:
- Huge penalty when losing powerups
- Limited number of in-flight bullets per player

**Comment by cxong** [comment]
![](http://img2.blogcu.com/images/o/v/e/overgamebatihan/1232989941gta1_pc_in_game_screenshot.png)

[GTA 1 & 2](http://en.wikipedia.org/wiki/Grand_Theft_Auto_%28video_game%29), the on-foot portions

Almost a different genre, more focus on the vehicle driving parts. The steering took a while to get used to, but worked surprisingly well. The on-foot sections were much less fun because aiming was hard (although there was autoaim) and you couldn't strafe.

**Comment by cxong** [comment]
![](https://upload.wikimedia.org/wikipedia/en/4/45/Mercs_Screenshot.png)

[Mercs](https://en.wikipedia.org/wiki/Mercs) (sequel to commando)

_Note that there are a few games like this: ikari warriors, gun smoke..._

Pros:
- Up to 3 player co-op
- Enemy grenades encourage movement
- Powerups like health

Cons:
- Repetitive

**Comment by cxong** [comment]
![](http://www.consoleclassix.com/info_img/Zombies_Ate_My_Neighbors_GEN_ScreenShot2.jpg)

[Zombies ate my neighbors](http://www.armchairempire.com/classic_gaming/zombies-ate-my-neighbors.htm)
- Interesting weapons (freeze gun, a gun that pushes the player back)

**Comment by cxong** [comment]
![](https://upload.wikimedia.org/wikipedia/en/5/5f/Smash_TV.png)

[Smash TV](http://www.mobygames.com/game/smash-tv) (also robotron)

Pros:
- Great control scheme that allows more intense battles
- Loads of pickups to encourage movement, create tension between getting powerups and staying in safe spot

Cons:
- Hard to fine tune the experience, need to think about crowd dynamics, powerups can be make-or-break
- Experience is either too easy or too confusing, hard to create a middle ground

**Comment by cxong** [comment]
![](https://upload.wikimedia.org/wikipedia/en/2/22/Gauntlet_screenshot.png)

[Gauntlet](http://www.mobygames.com/game/gauntlet)
- Classes
- Level design focuses action
- Health/time as a resource - health drains over time, food replenishes health

**Comment by cxong** [comment]
![](http://3.bp.blogspot.com/-LkePZoLkhek/TxYSRAefQ1I/AAAAAAAACNY/dNwXpzd1yXw/s1600/Alien_Syndrome_04.png)

[Alien Syndrome](http://redparsley.blogspot.com.au/2012/01/overhead-run-n-gun-games-7.html)

Pros:
- Great bosses
- Enemies move at random, harder to hit

Cons:
- Some backgrounds clash, making things hard to see

**Comment by cxong** [comment]
![](http://beefjack.com/files/2012/03/the-binding-of-isaac-wrath-of-the-lamb-gigantic-lasers-header.jpg)

[Binding of Isaac](http://store.steampowered.com/app/113200/)

Pros:
- Huge range of enemies, powerups, bosses
- Complex economy/inventory system with health, coins, keys, bombs etc.
- Randomly generated rooms and items (but this makes the pacing uneven)

Cons:
- Uneven variety in player "weapons"

**Comment by cxong** [comment]
![alien_breed](https://f.cloud.github.com/assets/1083215/572660/14029e22-c7a0-11e2-8479-753fd4d960cf.png)

Alien Breed

Pros:
- Great atmosphere, music
- Claustrophobic corridors, doors etc.
- Ammo/health/key pickups, and purchases
- Limited ammo plus constantly spawning enemies creates tension

Cons:
- Tension could be improved by using occlusion instead of having a small screen
- Apart from music, not much use of sound to create tension

**Comment by cxong** [comment]
![image](https://f.cloud.github.com/assets/1083215/1915815/da5f7774-7d6d-11e3-9052-f47b76c9ab68.png)

Crack Down
- Game mode: plant bombs and escape in allotted time
- Patrolling AI encourages stealthy, tactical gameplay
- Variety of terrain: lava, conveyor belts, electrified
- Can lie flat against wall to evade fire

**Comment by cxong** [comment]
![image](https://f.cloud.github.com/assets/1083215/1916722/f9b9bd66-7d7f-11e3-8f1d-e2304438df81.png)

The Ninja
- Upgradeable weapon (via drops from enemies)
- Hazards like falling boulders, floating logs
- Can block enemy shots with own shots
- Enemies that disguise as rocks, transform into wolves, teleport

**Comment by Bertram25** [comment]
This issue would be worth a blog post. :+1:

**Comment by cxong** [comment]
@Bertram25 I don't have a blog :smile: this issue is just a random collection of ideas anyway, maybe some time later I can distil some conclusions on what makes these games good

**Comment by Bertram25** [comment]
@cxong Would be worth knowing how you came to the conclusion one or another feature is good and why, indeed. :)

I guess on a release post in free game dev?
Best regards,

**Comment by cxong** [comment]
![image](https://f.cloud.github.com/assets/1083215/1927391/83254770-7e5a-11e3-8b8c-540ba599f24a.png)

[Vindicators](http://redparsley.blogspot.com.au/2012/03/overhead-run-n-gun-games-8.html)

The review of the game made an interesting point that although the level design encouraged tactical play, in practice it was frustrating because the game also had timing pressure with a constantly-depleting fuel gauge.

**Comment by cxong** [comment]
Hey @Bertram25 ; after 2 years I finally got around to that blog post :laughing: I couldn't cover all games listed here because then I'd be repeating myself a lot and I tried to add some chronological structure, but it still reads a bit like a brain dump. I hope you enjoy it anyway: http://cxong.github.io/2016/04/a-review-of-overhead-8-directional-shooters
